minerl/env/malmo.py

```diff
@@ -260,29 +265,30 @@ def _is_port_taken(port, address='0.0.0.0'):
             ports = [x.laddr.port for x  in psutil.net_connections()]
             return port in ports
 
-    @staticmethod
-    def _is_display_port_taken(port, x11_path):
-        return False
-
     @classmethod
     def _port_in_instance_pool(cls, port):
         # Ideally, this should be covered by other cases, but there may be delay
         # in when the ports get "used"
-        return port in [instance.port for instance in cls._instance_pool]
 
-    @classmethod
-    def configure_malmo_base_port(cls, malmo_base_port):
-        """Configure the lowest or base port for Malmo"""
-        cls._malmo_base_port = malmo_base_port
+        # shwang: Unfortunately, this check doesn't account for the multiprocess
+        # case where there are several instance pools. We leave it here despite
+        # this because it is a quick way to avoid expensively launching a
+        # second instance on the same port from the same process.
+        return port in [instance.port for instance in cls._instance_pool]
 
     @classmethod
     def _get_valid_port(cls):
-        malmo_base_port = cls._malmo_base_port
-        port = (cls.ninstances  % 5000) + malmo_base_port
-        while cls._is_port_taken(port) or \
-              cls._is_display_port_taken(port - malmo_base_port, cls.X11_DIR) or \
-              cls._port_in_instance_pool(port):
-            port += 1
+        # We might run into a race condition where another Minecraft process (or any
+        # process for that matter) binds to this port before our instance can
+        # do so. Randomization decreases the chance of that happening in
+        # between different MineRL InstanceManagers.
```

Comment:
At your best guess, how likely are we to hit an ephemeral port collision in this scheme?

Reply:
Pretty unlikely. If I saw this happening on the logs, I'd probably think it's a bug with the random seed or my computer is running a web server 😀